Added MK58 and SP Ammo to Sec Officer and Brigmedic loadouts

KarmaKitsuna opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description

Returned Security Officer and Brigmed sidearms
## Why we need to add this

SEC starts with side-arms on most other servers. Our SOP allows them to carry side-arms, but most armory's don't have enough. The station regularly goes on red alert so SEC can arm just to deal with vent critters or Lings.
## Media (Video/Screenshots)


## Checks

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**

:cl: KarmaKitsuna
- add: Sidearm loadout for Security Officer and Brigmedic.  Mk58 is the default but senior officers have the option of picking the Due Process. Both loadouts come with one magazine of SP ammo.

KarmaKitsuna committed:
Added MK58 and SP Ammo to Sec Officer and Brigmedic loadouts

PubliclyExecutedPig commented:
how does this work? Is it just another option is the security weapons tab of loadouts, or does it spawn no matter what you pick? I feel like even tho pistols are always better, there should be a choice

walksanatora commented:
Looking at the protos it just puts it in your bag if I am reading it right

KarmaKitsuna commented:
> how does this work? Is it just another option is the security weapons tab of loadouts, or does it spawn no matter what you pick? I feel like even tho pistols are always better, there should be a choice

This adds it to security officer and brigmedic bags roundstart. For reference I've seen other MRP servers do it this way, on Wizden they start with the pistol in their pocket. 

My concern with putting them as a weapon option for the loadout is that we still want all officers have a non-lethal side-arm. This isn't a choice between pistol or disabler, they get both. SOP and rules of escalation stay the same.

KillerTheFlareon commented:
> > how does this work? Is it just another option is the security weapons tab of loadouts, or does it spawn no matter what you pick? I feel like even tho pistols are always better, there should be a choice
> 
> This adds it to security officer and brigmedic bags roundstart. For reference I've seen other MRP servers do it this way, on Wizden they start with the pistol in their pocket.
> 
> My concern with putting them as a weapon option for the loadout is that we still want all officers have a non-lethal side-arm. This isn't a choice between pistol or disabler, they get both. SOP and rules of escalation stay the same.

Add the sidearm as a loadout option (like how the disablers have their own options) so in the future we can add more sidearms for security officers to choose between.

I am not saying make officers choose between the non-lethal or lethal option, but rather for officers to have an option to choose what KIND of lethal sidearm they get.

KarmaKitsuna commented:
> > > how does this work? Is it just another option is the security weapons tab of loadouts, or does it spawn no matter what you pick? I feel like even tho pistols are always better, there should be a choice
> > 
> > 
> > This adds it to security officer and brigmedic bags roundstart. For reference I've seen other MRP servers do it this way, on Wizden they start with the pistol in their pocket.
> > My concern with putting them as a weapon option for the loadout is that we still want all officers have a non-lethal side-arm. This isn't a choice between pistol or disabler, they get both. SOP and rules of escalation stay the same.
> 
> Add the sidearm as a loadout option (like how the disablers have their own options) so in the future we can add more sidearms for security officers to choose between.
> 
> I am not saying make officers choose between the non-lethal or lethal option, but rather for officers to have an option to choose what KIND of lethal sidearm they get.

That's a really good idea for future proofing. I'll change it

KarmaKitsuna committed:
Changed to allow for selection of side-arms in loadout menu

Happyrobot33 commented on Resources/Prototypes/Roles/Jobs/Security/security_officer.yml:
```diff
@@ -29,3 +29,4 @@
   storage:
     back:
     - Flash
+    - MagazinePistolSP
```
please make these part of the loadout itself, so different guns define different magazines

Happyrobot33 commented on Resources/Prototypes/Loadouts/Jobs/Security/security_officer.yml:
```diff
@@ -218,3 +218,10 @@
+  id: SecurityMk58
+  storage:
+    back:
+    - WeaponPistolMk58
```
Add the magazine here instead so different loadouts define different mags

KarmaKitsuna committed:
Added the Due Process as a side arm option for Senior Officers
Magazines for side arm loadouts are defined by the gun
Moved some code to the Starlight loadout folder, organized and commented on what is Starlight specific code

Happyrobot33 commented on Resources/Prototypes/Loadouts/role_loadouts.yml:
this file seems messed up. Please review to make sure its proper

KarmaKitsuna replied:
Can you give me a little more detail? What about it seems messed up? Should I not have the Starlight comments?

Happyrobot33 replied:
Nevermind i think i just read the changes wrong and didnt realize you moved brigmed

Happyrobot33 commented on Resources/Prototypes/Loadouts/Jobs/Security/security_officer.yml:
these to the starlight equivalent file

Happyrobot33 commented on Resources/Locale/en-US/preferences/loadout-groups.ftl:
move these to the starlight equivalent file

KarmaKitsuna committed:
Fixed the role_loadout file and moved loadout_groups changes to Starlight folder
Created a Security folder for Starlight specific job loadouts, added a security_officer.yml file to it and put changes in ther

KarmaKitsuna committed:
Added comments to role loadouts
